Source/CBL_Router.m

```diff
@@ -553,12 +553,12 @@ - (void) processRequestRanges {
     NSUInteger from, to;
     if (fromStr.length > 0) {
         from = (NSUInteger)fromStr.integerValue;
-        if (toStr.length > 0)
+        if (toStr.length > 0) {
             to = MIN((NSUInteger)toStr.integerValue, bodyLength - 1);
-        else
+            if (to < from)
+                return;  // invalid range
+        } else
```

Comment:
Yes, your are right. CocoaHTTPServer is not right for this case too. I will fix CocoaHTTPServer separately.